app/src/auth/auth.js

```diff
@@ -0,0 +1,96 @@
+var $ = require('jquery');
+var ko = require('knockout');
+var mapping = require('knockout-mapping');
+var config = require('../config');
+var sessionService = require('../services/session_service');
+
+// This is still the signin view model. There'll be another view model in this class.
+var AuthViewModel = function() {
+    var $authModal = $("#authModal");
+
+    var env = sessionStorage.getItem('environment');
```

Comment:
Create a persistent user options service for stuff like this. Abstract out sessionStorage.